src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -468,93 +513,97 @@ export default class RadixTransactionBuilder {
         return this
     }
 
+
     /**
      * Builds the atom, finds a node to submit to, adds network fee, signs the atom and submits
      * @param signer
      * @returns a BehaviourSubject that streams the atom status updates
      */
     public signAndSubmit(signer: RadixSignatureProvider) {
+        const atom = this.buildAtom()
+        
+        const stateSubject = new BehaviorSubject<string>('FINDING_NODE')
+
+        // Find a shard, any of the participant shards is ok
+        const shard = atom.getAddresses()[0].getShard()
+        
+        // Get node from universe
+        radixUniverse.getNodeConnection(shard)
+            .then(connection => {
+                RadixTransactionBuilder.signAndSubmitAtom(atom, connection, signer, this.participants.values())
+                    .subscribe(stateSubject)
+            })
+
+        return stateSubject
+    }
+
+    public buildAtom() {
         if (this.particleGroups.length === 0) {
-            throw new Error('No particles specified')
+            throw new Error('No particle groups specified')
         }
 
         const atom = new RadixAtom()
-
         atom.particleGroups = this.particleGroups
 
         // Add timestamp
         atom.setTimestamp(Date.now())
+        return atom
+    }
 
-        // Find a shard, any of the participant shards is ok
-        const shard = atom.getAddresses()[0].getShard()
-
-        // Get node from universe
-        let nodeConnection: RadixNodeConnection = null
-
-        const stateSubject = new BehaviorSubject<string>('FINDING_NODE')
-
+    public static signAndSubmitAtom(atom: RadixAtom, connection: RadixNodeConnection, signer: RadixSignatureProvider, participants: RadixAccount[]) {
         let signedAtom = null
+        
+        // Add POW fee
+        const endorsee = RadixAddress.fromPublic(connection.node.nodeInfo.system.key.bytes)
+
+        const stateSubject = new BehaviorSubject<string>('GENERATING_POW')
+        
+        RadixFeeProvider.generatePOWFee(
+            radixUniverse.universeConfig.getMagic(),
+            radixUniverse.powToken,
+            atom,
+            endorsee,
+        ).then(powFeeParticle => {
+            const powFeeParticleGroup = new RadixParticleGroup([RadixSpunParticle.up(powFeeParticle)])
+            atom.particleGroups.push(powFeeParticleGroup)
+
+            // Sign atom
+            stateSubject.next('SIGNING')
+            return signer.signAtom(atom)
+        }).then(_signedAtom => {
+            signedAtom = _signedAtom
+
+            logger.debug(signedAtom.hid.toString())
+
+            // Push atom into participant accounts to minimize delay
+            for (const participant of participants) {
+                participant._onAtomReceived({
+                    action: 'STORE',
+                    atom: signedAtom,
+                    processedData: {},
+                    isHead: true,
+                })
+            }
```

Comment:
Most likely I'm missing something, but is this really safe to do? Aren't we bypassing the ledger validation by "force feeding" the accounts like this? I would have expected the `_onAtomReceived` being called as a result of the ledger responding to a submission of an atom or so.

Reply:
Ah, wait! It's a "STORE" action, not "STORED", right?

Reply:
Nope, it's the same action it both cases. If the atom is rejected for any reason, we pass in a corresponding DELETE action. The higher up systems have no concept of different levels of validity of an atom.

It's not entirely safe, if you immediately start doing more stuff dependent on the atom you could run into trouble, but you could run into exactly the same kind of trouble even after receiving a STORED notification, should there be a consensus conflict down the line, though that's less likely of course.

I added this quite a long time ago to make the desktop wallet more responsive. The goal is to at some point push this down in to a separate Ledger layer that could manage things like submission state and finality more intelligently, but it's a hard problem.